cmd/swarmctl/service/flagparser/plugin.go

```diff
@@ -0,0 +1,18 @@
+package flagparser
+
+import (
+	"github.com/docker/swarmkit/api"
+	"github.com/spf13/pflag"
+)
+
+func parsePlugin(flags *pflag.FlagSet, spec *api.ServiceSpec) error {
```

Comment:
I kinda like the fact they all take the same signature (I think it was me originally doing this) - some functions need a service spec, some a task spec, some a container spec ... it's much easier on the caller.

Those parse functions could also be chained in a for loop